chainer/variable.py

```diff
@@ -1185,15 +1203,15 @@ def _to_device(self, device, allow_unchaining):
         elif not was_chainerx and is_chainerx:
             self._chainerx_name = self._node.name
 
+        # Note: It affects the whole behavior of Variable's methods.
+        #       We must place them in this place.
```

Comment:
Hmm, I don't get what this comment is trying to convey.

Reply:
`self._device` and `self._has_chainerx_array` is referenced in many places both explicitly and implicitly. Reassining them leads to change the behavior of some methods and properties. For example, `grad_var = self.grad_var` doesn't work as expected if we place them before it.

Reply:
I don't see any particular reason to make such comment to these lines.
It's in fact very common that timing of attribute assignment matters.

Reply:
Hmm, OK. I'll remove it.